### DISCUSSION TITLE
Determine locale on server

### DISCUSSION THREAD

**Discussion by raphi-0901** [nuxt-modules/i18n#3044] [Q&A]
I have a server route and want to return an object where I need the locale which was also determined by the localeDetector:

Just like this:
```js
export default defineEventHandler(async(event) => {
  const t = await useTranslation(event)

  return {
    message: t("message"),
    locale: ???
  }
})
```

How can I get the locale?

**Comment by raphi-0901** [comment]
Well what just came to my mind is that I could define a key in my translation files which point to the corresponding locale.

en.json:
```json
{
  "locale": "en"
}
```

de.json:
```json
{
  "locale": "de"
}
```

and then simply get it like this:
```js
const t = await useTranslation(event)
const locale = t('locale')
```

**Comment by BobbieGoede** [comment]
I think you should be able to access the detected locale with `event.context.i18n?.locale`
